File: pinyougou-sellergoods-interface/src/main/java/com/lsf/pinyougou/sellergoods/service/GoodsService.java
package com.lsf.pinyougou.sellergoods.service;

import java.util.List;

import com.lsf.pinyougou.pojo.TbGoods;
import com.lsf.pinyougou.pojogroup.Goods;
import vo.PageResult;


/**
 * 商品服务层接口
 */
public interface GoodsService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbGoods> findAll();


    /**
     * 无条件分查询
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 多条件分页查询
     *
     * @param pageNum  当前页
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPageLimit(TbGoods goods, int pageNum, int pageSize);


    /**
     * 添加商品
     */
    void add(Goods goods);


    /**
     * 修改商品信息
     */
    void update(Goods goods);


    /**
     * 根据商品 ID 查询商品信息，返回商品组合实体类对象 Goods
     *
     * @param id
     * @return
     */
    Goods findOne(long id);


    /**
     * 批量删除商品，只进行逻辑删除
     *
     * @param ids
     */
    void batchDelete(Long[] ids);


    /**
     * 修改商品的审核状态
     */
    void updateGoodStatus(Long[] ids, String status);


    /**
     * 修改商品的上下架状态
     *
     * @param ids
     * @param status
     */
    void updateGoodMarketable(Long[] ids, String status);
}

File: pinyougou-shop-web/src/main/webapp/js/service/goodsService.js
// 服务层
pyg.service('goodsService', function ($http) {

    // 查询所有数据
    this.findAll = function () {
        return $http.get('../goods/findAll.do');
    };

    // 无条件分页查询
    this.findPage = function (page, size) {
        return $http.get('../goods/findPage.do?page=' + page + '&size=' + size);
    };

    // 根据 ID 查询实体
    this.findOne = function (id) {
        return $http.get('../goods/findOne.do?id=' + id);
    };

    // 添加商品
    this.add = function (entity) {
        return $http.post('../goods/add.do', entity);
    };

    // 修改
    this.update = function (entity) {
        return $http.post('../goods/update.do', entity);
    };

    // 批量删除
    this.batchDelete = function (selectedList) {
        return $http.get('../goods/batcheDelete.do?ids=' + selectedList);
    };

    // 多条件分页查询
    this.findPageLimit = function (page, size, searchEntity) {
        return $http.post('../goods/search.do?page=' + page + "&size=" + size, searchEntity);
    };
});
